solutions/CSES Subarray Sums II.mdx

```diff
@@ -0,0 +1,175 @@
+const int INF = 1e9;
+const ll LINF = 1e18;
+const ld PI = acos((ld)-1);
+const int dx[4] = {1, 0, -1, 0}, dy[4] = {0, 1, 0, -1};
+//mt19937 rng((uint32_t)chrono::steady_clock::now().time_since_epoch().count()); 
+
+namespace input {
+    template<class T> void re(complex<T>& x);
+    template<class T1, class T2> void re(pair<T1,T2>& p);
+    template<class T> void re(vector<T>& a);
+    template<class T, size_t SZ> void re(array<T,SZ>& a);
+
+    template<class T> void re(T& x) { cin >> x; }
+    void re(double& x) { string t; re(t); x = stod(t); }
+    void re(ld& x) { string t; re(t); x = stold(t); }
+    template<class Arg, class... Args> void re(Arg& first, Args&... rest) { 
+        re(first); re(rest...); 
+    }
+
+    template<class T> void re(complex<T>& x) { T a,b; re(a,b); x = cd(a,b); }
+    template<class T1, class T2> void re(pair<T1,T2>& p) { re(p.f,p.s); }
+    template<class T> void re(vector<T>& a) { F0R(i,sz(a)) re(a[i]); }
+    template<class T, size_t SZ> void re(array<T,SZ>& a) { F0R(i,SZ) re(a[i]); }
+}
+
+using namespace input;
+
+namespace output {
+    template<class T1, class T2> void pr(const pair<T1,T2>& x);
+    template<class T, size_t SZ> void pr(const array<T,SZ>& x);
+    template<class T> void pr(const vector<T>& x);
+    template<class T> void pr(const set<T>& x);
+    template<class T1, class T2> void pr(const map<T1,T2>& x);
+
+    template<class T> void pr(const T& x) { cout << x; }
+    template<class Arg, class... Args> void pr(const Arg& first, const Args&... rest) { 
+        pr(first); pr(rest...); 
+    }
+
+    template<class T1, class T2> void pr(const pair<T1,T2>& x) { 
+        pr("{",x.f,", ",x.s,"}"); 
+    }
+    template<class T> void prContain(const T& x) {
+        pr("{");
+        bool fst = 1; for (const auto& a: x) pr(!fst?", ":"",a), fst = 0; // const needed for vector<bool>
+        pr("}");
+    }
+    template<class T, size_t SZ> void pr(const array<T,SZ>& x) { prContain(x); }
+    template<class T> void pr(const vector<T>& x) { prContain(x); }
+    template<class T> void pr(const set<T>& x) { prContain(x); }
+    template<class T1, class T2> void pr(const map<T1,T2>& x) { prContain(x); }
+    
+    void ps() { pr("\n"); }
+    template<class Arg> void ps(const Arg& first) { 
+        pr(first); ps(); // no space at end of line
+    }
+    template<class Arg, class... Args> void ps(const Arg& first, const Args&... rest) { 
+        pr(first," "); ps(rest...); // print w/ spaces
+    }
+}
+
+using namespace output;
+
+int N, X;
+vi T;
+
+int main() {
+    ios_base::sync_with_stdio(0);
+    cin.tie(0); cout.tie(0);
+
+    cin >> N >> X;
+    T.resize(N); re(T);
+    
+    ll psum = 0;
+    ll res = 0;
+    map<ll, int> sums;
+    sums[0]=1;
+
+    trav(x, T){
+        psum += x;
+        
+        //If there is a subarray with psum-x, we can exclude from our current subarray. Thus, we can add the number of those subarrays to our answer.
+        if(sums[psum-X] != 0){
```

Comment:
not sure why you have this if statement, this will add `psum-X` to the map anyways.

Reply:
fixed, thx for pointing that out! ig it didn't come to mind when i was programming it